internal/host/hostutil/update_host.go

```diff
@@ -68,12 +68,15 @@ func UpdateHostStatus(ctx context.Context, log logrus.FieldLogger, db *gorm.DB,
 			hostId, infraEnvId, srcStatus, newStatus)
 	}
 
+	if host, err = common.GetHostFromDB(db, infraEnvId.String(), hostId.String()); err != nil {
+		return nil, errors.Wrapf(err, "failed to read from host %s from infraEnv %s from the database after the update", hostId, infraEnvId)
+	}
+
 	if newStatus != srcStatus {
 		if statusInfo != "" {
 			statusInfo += fmt.Sprintf("(%s)", statusInfo)
 		}
-		// TODO Need event for infra-env instead of cluster
-		eventgen.SendHostStatusUpdatedEvent(ctx, eventsHandler, hostId, infraEnvId, GetEventSeverityFromHostStatus(newStatus),
+		eventgen.SendHostStatusUpdatedEvent(ctx, eventsHandler, hostId, infraEnvId, host.ClusterID, GetEventSeverityFromHostStatus(newStatus),
```

Comment:
does this code know how to handle nil cluster id?

Reply:
yes, the property for the cluster_id is defined as: `cluster_id: UUID_PTR` which assumes it might be nil and treat it accordingly.

Reply:
Yup.
This is why when I added `cluster_id` as `UUID_PTR`, it adds it to `internal/common/events/events.go` as `*strfmt.UUID`

Reply:
some UTs do assert for an actual cluster_id, while others assert for an empty one.